src/pkgcheck/checks/python.py

```diff
@@ -80,6 +100,36 @@ def get_python_eclass(pkg):
         assert(len(eclasses) <= 1)
         return eclasses.pop() if eclasses else None
 
+    def scan_tree_recursively(self, deptree, expected_cls):
```

Comment:
As we discussed on briefly on IRC, I think functionality for this should be added to relevant classes in pkgcore. I can possibly look into that this coming week.